test_e2e.sh

```diff
@@ -1072,6 +1074,81 @@ function e2e::exechook_fail_retry() {
     fi
 }
 
+##############################################
+# Test exechook-success with GIT_SYNC_ONE_TIME
+##############################################
+function exechook_success_once_helper() {
+    # First sync
+    echo "$FUNCNAME 2" > "$REPO"/file
+    git -C "$REPO" commit -qam "$FUNCNAME 2"
+
+    GIT_SYNC \
+        --period=${2}ms \
+        --one-time \
+        --repo="file://$REPO" \
+        --branch="$MAIN_BRANCH" \
+        --root="$ROOT" \
+        --link="link" \
+        --exechook-command="$EXECHOOK_COMMAND_SLEEPY" \
+        >> "$1" 2>&1 &
+    sleep 7
+    assert_link_exists "$ROOT"/link
+    assert_file_exists "$ROOT"/link/file
+    assert_file_exists "$ROOT"/link/exechook
+    assert_file_exists "$ROOT"/link/link-exechook
+    assert_file_eq "$ROOT"/link/file "$FUNCNAME 2"
+    assert_file_eq "$ROOT"/link/exechook "$FUNCNAME 2"
+    assert_file_eq "$ROOT"/link/link-exechook "$FUNCNAME 2"
+}
+
+function e2e::exechook_success_once() {
+  for i in $(seq 0 50 200); do
```

Comment:
Why do we need this in a loop?  It should be sufficient to demonstrate that a slow hook actually completed one time (that main waitied for it), right?

Reply:
Hmm, this makes sense. My reason for running the same test several times was to be more sure of the results. Since race conditions may or may not be detected due to the OS' scheduling of process/threads, I thought we needed multiple test runs to be more sure of the results.

However, we could never really be 100% sure using this approach anyway, so I think we may be good with running each test exactly once, and maybe making the sleep last for a bit :)

Reply:
In this case, I don't think we need the helper functions really, so I'll go ahead and delete them.